homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -5,52 +5,110 @@
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
     CONF_ADDRESS,
     CONF_DEVICE,
     CONF_NAME,
+    CONF_PORT,
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
     POWER_WATT,
+    TEMP_CELSIUS,
 )
+from homeassistant.exceptions import InvalidStateError
 import homeassistant.helpers.config_validation as cv
 
+from .aurora_device import AuroraDevice
+from .const import (
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_ADDRESS,
+    DOMAIN,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_ADDRESS = 2
-DEFAULT_NAME = "Solar PV"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_DEVICE): cv.string,
         vol.Optional(CONF_ADDRESS, default=DEFAULT_ADDRESS): cv.positive_int,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME, default="Solar PV"): cv.string,
     }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Aurora ABB PowerOne device."""
+    """Set up single sensor based on configuration.yaml (DEPRECATED)."""
     devices = []
     comport = config[CONF_DEVICE]
-    address = config[CONF_ADDRESS]
-    name = config[CONF_NAME]
-
+    address = config.get(CONF_ADDRESS, DEFAULT_ADDRESS)
+    name = config.get(CONF_NAME, "Solar PV")
+    _LOGGER.warning(
+        "DEPRECATED: setting up %s via configuration.yaml will "
+        "soon be unsupported.  Please remove the entries from your "
+        "configuration.yaml file and then set up the integration via the UI",
+        DOMAIN,
+    )
     _LOGGER.debug("Intitialising com port=%s address=%s", comport, address)
     client = AuroraSerialClient(address, comport, parity="N", timeout=1)
 
-    devices.append(AuroraABBSolarPVMonitorSensor(client, name, "Power"))
+    config.data = {
+        "device_name": name,
+        ATTR_SERIAL_NUMBER: None,
+        ATTR_FIRMWARE: None,
+        ATTR_MODEL: None,
+    }
+    config.entry_id = "undefined_entry_id"
+    devices.append(AuroraSensor(client, config, "Power", "instantaneouspower"))
     add_entities(devices, True)
 
 
-class AuroraABBSolarPVMonitorSensor(SensorEntity):
-    """Representation of a Sensor."""
+async def async_setup_entry(hass, config, async_add_entities) -> None:
+    """Set up aurora_abb_powerone sensor based on a config entry."""
+    entities = []
+
+    comport = config.data[CONF_PORT]
+    address = config.data[CONF_ADDRESS]
+    client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+
+    sensortypes = [
+        {"parameter": "instantaneouspower", "name": "Power Output"},
+        {"parameter": "temperature", "name": "Temperature"},
+    ]
+
+    for sens in sensortypes:
+        entities.append(AuroraSensor(client, config, sens["name"], sens["parameter"]))
+
+    hass.data[DOMAIN][config.entry_id] = {"client": client}
```

Comment:
Normally we setup the device in `__init__.py`'s `async_setup_entry` so we can raise `ConfigEntryNotReady` if the device is offline which allows setup to be retired later.

Reply:
Ok. Done.